lib/libc/minimal/source/string/string.c

```diff
@@ -170,6 +170,37 @@ int strncmp(const char *s1, const char *s2, size_t n)
 	return (n == 0) ? 0 : (*s1 - *s2);
 }
 
+/**
+ * @brief Separate `str` by any char in `sep` and return NULL terminated
+ * sections. Consecutive `sep` chars in `str` are treated as a single
+ * separator.
+ *
+ * @return pointer to NULL terminated string or NULL on errors.
+ */
+char *strsep(char **str, const char *sep)
+{
+	char *start, *end;
+
+	if (str == NULL || *str == NULL || *str[0] == '\0') {
```

Comment:
`strsep` on Linux segfaults if `str` is a null pointer: that is an API mis-use.  It should probably do the same here.

Also it does not return null if `*str[0]` is zero: it returns the original value of `*str` and sets `*str` to NULL.

Reply:
I'll make sure new strok_t implementation complies with the specs. Thanks for the review.

Reply:
You're welcome.  I do think `strtok_r` will be a good enhancement, and look forward to seeing that available.